lib/repl.js

```diff
@@ -1256,13 +1256,16 @@ function defineDefaultCommands(repl) {
       try {
         var stats = fs.statSync(file);
         if (stats && stats.isFile()) {
+          this.editorMode = true;
+          REPLServer.super_.prototype.setPrompt.call(this, '');
```

Comment:
Yes, and it duplicates the behavior in the `.editor` [command](https://github.com/nodejs/node/blob/master/lib/repl.js#L1282). As I was working on this, I thought this could benefit from a little refactoring so that there is an `enableEditorMode()` function that handles everything in one place. Do you think it's needed?